### PR TITLE
Create Homework_3.py

### PR DESCRIPTION
**Pull request by Verantina007**

### CODE REVIEW

File: Homework_3.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	tHis iz your homeWork, copy these Text to variable. 
+
+	You NEED TO normalize it fROM letter CASEs point oF View. also, create one MORE senTENCE witH LAST WoRDS of each existING SENtence and add it to the END OF this Paragraph.
+
+	it iZ misspeLLing here. fix“iZ” with correct “is”, but ONLY when it Iz a mistAKE. 
+
+	last iz TO calculate nuMber OF Whitespace characteRS in this Text. caREFULL, not only Spaces, but ALL whitespaces. I got 87.
+'''
+
+import re # library for useing regex
```
Comment (romkabauer) [inline Homework_3.py]: I suggest bringing all imports up to the top according to pep8 standard (ref: https://peps.python.org/pep-0008/#imports)

File: Homework_3.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+normilized_str = '' #new empty string for normilized string
+
+for element in input_str.lower().split('.'): # circle for list of sentences and make all elements in lower case
+    if re.search(r'[a-z]', element): # find the first letter in each sentence
+        # make firest letter in each sentence capitalized and reasign element value
+        element = element[0:re.search(r'[a-z]', element).start()]+\
+                  element[re.search(r'[a-z]', element).start()].capitalize()+\
+                  element[re.search(r'[a-z]', element).start()+1:]
+        normilized_str += element + '.' # add reasigned element to normilized string and dot at the end of each sentence
+    else:
```
Comment (romkabauer) [inline Homework_3.py]: Why not skip empty lines?:)

File: Homework_3.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import re # library for useing regex
+
+# 1. Normalize the text
+
+normilized_str = '' #new empty string for normilized string
+
+for element in input_str.lower().split('.'): # circle for list of sentences and make all elements in lower case
+    if re.search(r'[a-z]', element): # find the first letter in each sentence
+        # make firest letter in each sentence capitalized and reasign element value
+        element = element[0:re.search(r'[a-z]', element).start()]+\
```
Comment (romkabauer) [inline Homework_3.py]: Maybe use `.strip()` to get rid of leading spaces and capitalize the whole string with the sentence?

File: Homework_3.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	last iz TO calculate nuMber OF Whitespace characteRS in this Text. caREFULL, not only Spaces, but ALL whitespaces. I got 87.
+'''
+
+import re # library for useing regex
+
+# 1. Normalize the text
+
+normilized_str = '' #new empty string for normilized string
+
+for element in input_str.lower().split('.'): # circle for list of sentences and make all elements in lower case
```
Comment (romkabauer) [inline Homework_3.py]: I suggest replacing `element` with more meaningful name like `sentence` to make code more readable

File: Homework_3.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        normilized_str += '\n' # add element if there are no letters in the sentence
+
+print ('1. Normilizwd string is: \n', normilized_str) # print the normilized sentence
+
+# 2. Create a new sentence with last words of each existing sentence and add it at the end of the text
+
+new_sentence = '' # empty string for new sentence
+
+for element in input_str.lower().split('.'): # circle for list of sentences and make all elements in lower case
+    if re.search(r'[a-z]', element): # find sentences with words
```
Comment (romkabauer) [inline Homework_3.py]: Firstly, why not clean the text from empty lines?:) And, secondly, I think if you search for "words" in the sentence, it is better to use regexp token `\w` (may be helpful: https://regex101.com/)

File: Homework_3.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        new_sentence += element.split()[-1] + ' ' # add last word from each sentence in new_sentence
+
+str_with_new_sentence = normilized_str + ' ' + new_sentence[0].capitalize() + new_sentence[1:] # add new sentence in the input string
+print ('2. String with new sentence is: \n', str_with_new_sentence) # print string with new sentence
```
Comment (romkabauer) [inline Homework_3.py]: According to the task, it is necessary to add new sentence to the end of the 3rd paragraph, not in the end of the text:) And, please add point to the end of this sentence:)

File: Homework_3.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+#3. Fix 'IS' misspelling
+#while re.findall(r' iz ', str_with_new_sentence, re.I):
+
+fixed_string = re.sub(re.compile(' iz ', re.I), ' is ', str_with_new_sentence) # find all ' iz ' ignoring case and replace it with ' is '
+print('3. Fixed string is: \n', fixed_string) # print fixed string
+
+# 4. Count the number of whitespaces
+n = len(re.findall(r'[ \t\n\r\f\v]', input_str)) # find all whitespaces and count the length of list with it
```
Comment (romkabauer) [inline Homework_3.py]: Token `\s` exists in regexp to find white spaces of any kind:) (ref: https://regex101.com/)

**Review by romkabauer — CHANGES_REQUESTED** [review]

File: Homework_3.py (hunk 1 above)
Comment (Verantina007) [reply Homework_3.py]: Done

File: Homework_3.py (hunk 2 above)
Comment (Verantina007) [reply Homework_3.py]: Done

File: Homework_3.py (hunk 4 above)
Comment (Verantina007) [reply Homework_3.py]: Done

File: Homework_3.py (hunk 3 above)
Comment (Verantina007) [reply Homework_3.py]: Done

File: Homework_3.py (hunk 5 above)
Comment (Verantina007) [reply Homework_3.py]: Done

File: Homework_3.py (hunk 6 above)
Comment (Verantina007) [reply Homework_3.py]: Done

File: Homework_3.py (hunk 7 above)
Comment (Verantina007) [reply Homework_3.py]: Done

**Review by romkabauer — APPROVED** [review]

**Pull request merged**